deliaioanac/assignment2/LCA.cpp

```diff
@@ -0,0 +1,157 @@
+#include <iostream>
+#include <vector>
+#include <math.h>
+#include <stdio.h>
+#include <utility>
+#include <stdlib.h>
+
+
+// Structure for a tree node
+struct TreeNode {
+    int val;
+    std::vector<TreeNode*> children;
+    TreeNode(int x) : val(x) {}
+    ~TreeNode() {
+        while (!children.empty()) {
+            delete children.back();
+            children.pop_back();
+        }
+    }
+};
+
+// Structure for tree representation
+class CustomTree {
+    private:
+        TreeNode* tree;
+        std::vector<int> first, eulerNodes, eulerLevels;
+        std::vector<std::vector<int> > rmq;
+        void makeEuler(TreeNode *node, int level);
+        void makeRMQ();
+    public:
+        void createTree(const std::vector<int> &parents, int &k, TreeNode *node = nullptr);
+        void destroyTree();
+        void preprocessingLCA();
+        int getLowestCommonAncestor(int node1, int node2);
+};
```

Comment:
The same comment as in printAncestors.cpp about constructors/destructors & naming :)